etl/steps/data/garden/itopf/2023-05-18/oil_spills.py

```diff
@@ -0,0 +1,87 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+import numpy as np
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("oil_spills.start")
+
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow: Dataset = paths.load_dependency("oil_spills")
+
+    # Read table from meadow dataset.
+    tb_meadow = ds_meadow["oil_spills"]
+    # Convert the table 'tb_meadow' into a pandas DataFrame.
+    df = pd.DataFrame(tb_meadow)
+
+    # Log that the process of harmonizing countries has started.
+    log.info("oil_spills.harmonize_countries")
+    # Call a function to harmonize country names in the DataFrame.
+    df = geo.harmonize_countries(
+        df=df, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path)
+
+    # Create an empty DataFrame to store the average values per decade.
+    decadal_averages_df = pd.DataFrame()
+
+    # Iterate over the columns of the original DataFrame.
+    for column in df.columns:
+        # Select specific columns of interest.
+        if column in ['bel_700t', 'ab_700t', 'oil_spilled']:
+            # Group the data by decade (dividing the year by 10, rounding down and multiplying by 10 again),
+            # and calculate the mean for the specified column. Drop NaN results.
+            decadal_averages = df.groupby(df['year'] // 10 * 10)[column].mean().dropna()
+
+            # Construct new column names for the decadal averages.
+            decadal_column = 'decadal_' + str(column)
+
+            # Add the calculated decadal averages to the new DataFrame, rounding and converting to integers.
+            decadal_averages_df[decadal_column] = np.round(decadal_averages).astype(int)
+
+    # Merge the original DataFrame with the DataFrame containing the decadal averages,
+    # using 'year' as the key and keeping all records from both DataFrames ('outer' join).
+    df_decadal = pd.merge(df, decadal_averages_df, on='year', how='outer')
+
+    # Set the DataFrame index to be a combination of 'country' and 'year' and check if the index is unique.
+    df_decadal.set_index(['country', 'year'], inplace = True)
+    assert df_decadal.index.is_unique, "Index is not unique."
+
+    # Reset the DataFrame's index to default.
+    df_decadal.reset_index(inplace = True)
+
+    # Replace any '__' in column names with a space.
+    newnames = [name.replace('__', ' ') for name in df_decadal.columns]
```

Comment:
oh this is because I actually have double __ in some column names so I was just converting it to only have _ (will take a look why this happens for only some of my column names sometimes though